setup.py

```diff
@@ -99,6 +115,48 @@ def run(self) -> None:
         cmake_cmd = ["cmake", "--build", "."] + self.build_args
         subprocess.run(cmake_cmd, cwd=self.build_temp, check=True)
 
+        if editable_mode:
+            # NOTE(brad) When installed in editable mode, python expects to find symengine
+            # in the symenginepy/symengine source directory. Everything is already there
+            # except the compiled symengine_wrapper extension module, so we copy that there
+            # as well.
+            # NOTE(brad) For some reason that I don't fully understand, the symengine_wrapper
+            # shared library is neither present in symengine_install nor needed in the source
+            # directory on macos sometimes (for example, on the github actions macos runner).
+            # So, if the copy fails (due to the file not being present, we just move on).
```

Comment:
I'm still a little sketched out by this but I'm on board especially since this is seemingly better than what we have now - ok to merge and punt on figuring this out

Reply:
Yeah. I feel like this whole solution is kind of sketch.

For a very long time, I was completely mystified by what was going wrong. This is the first working solution I came across.

Now, I kind of wonder if a better angle would be to preserve the whole build directory. Like, perform the build in `SOURCE_DIR/build`, then copy it into the temporary directory pip and setuptools expect it to be in (so that none of their stuff gets broken, I don't know what sorts of assumptions they make).

Then we delete everything where we were copying `cc_sym` and the like into the SOURCE_DIR. The point of this is to make it so that when `setuptools` calls `find_package`, it only finds the one in `SOURCE_DIR/build`. I think if that happens, the editable install will expect to find `cc_sym` in `build/pybind`. More over, since the build was already there, the runpaths will be correct by default.

Though, I am mildly concerned by `symengine_wrapper`, as I'm not sure if the fact that we set
``` python
    package_dir={
        "symforce": "symforce",
        "symengine": "third_party/symenginepy/symengine",
    },
```

will prevent setuptools from looking for `symengine` in the build directory. So maybe that copying might still have to be done until I come up with a better solution.

Anyway, while I have ideas, I also think it would be preferable to make symforce editable installable for users in the meantime.

Reply:
I'm pretty ok with this approach as-is in general, specifically I was highlighting the second NOTE here - it seems weird that sometimes on macos, symengine_wrapper isn't where we expect it to be and doesn't need to be and instead ends up who-knows-where but works anyway.  I'm ok not investigating further, but it might be good to at least figure out where the library is ending up